Initial commit

fedaktaras opened the pull request:
It seems to be okay, but there is a chance to get the same color and number for the ball two or even three times in a row.

fedaktaras committed:
Initial commit

karma-o commented on src/main/java/core/basesyntax/Application.java:
```diff
@@ -3,5 +3,9 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        System.out.println("First " + lottery.getRandomBall());
+        System.out.println("Second " + lottery.getRandomBall());
+        System.out.println("Third " + lottery.getRandomBall());
```
let's generate and sout balls in a loop

karma-o commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,11 @@
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        Random random = new Random();
```
make random a class-level var

karma-o commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,11 @@
-        return null;
+        Random random = new Random();
+        Color someColor = Color.values()[random.nextInt(Color.values().length)];
+        return someColor.toString();
```
use enum's `name()` instead of `toString()` try to think why it would be better 😉

fedaktaras replied:
toString() can be overridden name() is final and it is ensuring that it will always return the exact name of the enum constant as declared in the enum type. Better to read code. The explanations I found.

karma-o commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,14 @@
+public class Lottery {
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
```
let's also avoid the creation of new `ColorSupplier` and `Random` each time this method is called

karma-o commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,14 @@
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100));
```
magic number should be a constant
Also a bound parameter of `nextInt()` is exclusive, so we should account for that, check Javadoc of the `nextInt()` for details

karma-o reviewed (changes requested):
Please check common mistakes in the task description before submitting your solution for review, next time it won't be checked until they are fixed

fedaktaras committed:
Refactored code according to mentor's suggestions
No magic numbers
Level class variables where it is necessary
random.nextInt() + 1 for numbers from 1
enum. name() is better to show enum's name.

yevheniisynytsia reviewed (changes requested):
Well done! Left some comments to work on

fedaktaras commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,16 @@
+
     public String getRandomColor() {
-        return null;
+        Color someColor = Color.values()[random.nextInt(Color.values().length)];
```
Agree. It is much better.

fedaktaras committed:
Refactored code according to mentor's suggestions
- Erasing magic numbers missed made one constant. Fixed.
- In an Application class moved some variables to class level.
- Typo corrections. (++i -> i++)
- Variable name corrections. (someColor -> randomColor)
- Style corrections.

andrii-hoienko reviewed (changes requested):
Nice, but some improvements required

fedaktaras committed:
Refactored code according to mentor's suggestions
final static proper naming
moved NUMBER_MAX_BOUNDARY to class level and proper naming

andrii-hoienko reviewed (approved):
Gj
